packages/api/src/record.ts

```diff
@@ -304,14 +444,55 @@ export class Record implements RecordModel {
    *
    * @beta
    */
-  async send(target: string): Promise<ResponseStatus> {
-    const { reply: { status } } = await this._agent.sendDwnRequest({
-      messageType    : DwnInterfaceName.Records + DwnMethodName.Write,
-      author         : this._connectedDid,
-      dataStream     : await this.data.blob(),
-      target         : target,
-      messageOptions : this.toJSON(),
-    });
+  async send(_options?: string | { [key: string]: any }): Promise<ResponseStatus> {
+
+    const initialWrite = this._initialWrite;
+    const options = !_options ? { target: this._connectedDid } : typeof _options === 'string' ? { target: _options } : _options;
+
+    if (!options.target){
+      options.target = this._connectedDid;
```

Comment:
This is some ugly looking code 😄 

It seems like there should be a better way to accomplish this.

Reply:
Liran changed it